web/Services/AzureStorage.cs

```diff
@@ -0,0 +1,124 @@
+    {
+        private readonly string _storageConnectionString;
+        private readonly string _baseContainerName;
+        private readonly ILogger<AzureStorage> _logger;
+
+        public AzureStorage(IConfiguration configuration, ILogger<AzureStorage> logger)
+        {
+            _storageConnectionString = configuration.GetValue<string>("BlobConnectionString");
+            _baseContainerName = configuration.GetValue<string>("BlobContainerName");
+            _logger = logger;
+        }
+
+        private string GetContainerName(string varType, int id)
+        {
+            return _baseContainerName + varType + id.ToString();
+        }
+
+        private async Task<BlobContainerClient> GetContainerClient(string varType, int id)
+        {
+            string containerName = GetContainerName(varType, id);
+            BlobContainerClient containerClient = new BlobContainerClient(_storageConnectionString, containerName);
+            if (!containerClient.Exists())
+            {
+                await CreateContainerAsync(varType, id);
+                containerClient = new BlobContainerClient(_storageConnectionString, containerName);
+            }
+            return containerClient;
+        }
+
+        public async Task CreateContainerAsync(string varType, int id)
+        {
+            string containerName = GetContainerName(varType, id);
+            BlobServiceClient blobServiceClient = new BlobServiceClient(_storageConnectionString);
+            var container = blobServiceClient.GetBlobContainerClient(containerName);
+            if (container.Exists())
+            {
+                await DeleteContainerAsync(varType, id);
+            }
+            await blobServiceClient.CreateBlobContainerAsync(containerName);
+        }
+        
+        public async Task DeleteContainerAsync(string varType, int id)
+        {
+            var containerClient = await GetContainerClient(varType, id);
+            await containerClient.DeleteAsync();
+        }
+
+        public async Task UploadFileAsync(IFormFile blob, string varType, int id)
+        {
+            var containerClient = await GetContainerClient(varType, id);
+            var blobClient = containerClient.GetBlobClient(blob.FileName);
+            int i = 0;
+            while (blobClient.Exists())
+            {
+                i += 1;
+                string fileName = SplitFileName(blob.FileName).name + i + "." + SplitFileName(blob.FileName).type;
+                blobClient = containerClient.GetBlobClient(fileName);
+            }
+            await using (Stream? data = blob.OpenReadStream())
+            {
+                await blobClient.UploadAsync(data);
+            }
+        }
+
+        public async Task<List<string>> ListFilesAsync(string varType, int id)
+        {
+            var containerClient = await GetContainerClient(varType, id);
+            List<string> fileNames = new List<string>();
+            fileNames.Add(id.ToString());

```

Comment:
Is this intentional, add the id as a file name?

Reply:
It was a way to allow the property Id to be accessed on the Image List page (so that new images can be added to the correct property) - there might be a way to do this using the routing, but adding the id to the start of the file list seemed more straightforward.

Reply:
I think it would be nicer to have a view model class that has the property id and a list of files. That would be easier for someone else reading the code (me included) to quickly understand.